core/ide/src/main/java/org/alice/ide/ProjectApplication.java

```diff
@@ -386,100 +390,136 @@ private final void loadProject(UserActivity activity, UriProjectLoader uriProjec
+        case YES -> {
+          // load a backup
+          loadProject(newProjectActivity(), new FileProjectLoader(backup, false), true, true, new HashSet<>());
 
-      if (parentDir != null) {
-        isBackup = BACKUP_EXTENSION.equals(getExtension(parentDir.getName()));
-      }
+          yield true;
+        }
+        case NO -> {
+          // discard the backups
+          try {
+            FileUtils.deleteDirectory(backupDir);
+          } catch (IOException e) {
+            Logger.throwable(e, "Unable to delete default backup directory.");
+          }
+
+          yield false;
+        }
+        case CANCEL ->
+          // unreachable path
+          false;
+      };
+    }
+
+    return false;
+  }
+
+  private void projectLoaded(UserActivity activity, Project project, boolean isLoadingBackups,
+                             boolean isMainProjectCorrupted, Set<String> unloadableFiles) {
+    File saved = UriUtilities.getFile(getUri());
+
+    if (saved != null && !projectFileUtilities.isProject(saved)) {
+      return;
     }
 
+    boolean isBackup = uriProjectLoader.isBackup();
+
     if (project == null) {
-      handleProjectLoadError(saved, activity, isBackup, isLoadingBackups, unloadableFiles);
+      handleProjectLoadError(saved, activity, isBackup, isLoadingBackups, isMainProjectCorrupted, unloadableFiles);
     } else {
-      handleProjectLoadSuccess(project, saved, activity, isBackup, isLoadingBackups, unloadableFiles);
+      handleProjectLoadSuccess(project, saved, activity, isBackup, isLoadingBackups, isMainProjectCorrupted, unloadableFiles);
     }
   }
 
   private void handleProjectLoadError(File projectFile, UserActivity activity, boolean isBackup,
-                                      boolean isLoadingBackups, Set<String> unloadableFiles) {
-    File backupDir = projectFileUtilities.backupDirectory(projectFile, isBackup).toFile();
+                                      boolean isLoadingBackups, boolean isMainProjectCorrupted,
+                                      Set<String> unloadableFiles) {
+    File backupDir = projectFileUtilities.appropriateBackupDirectory(projectFile).toFile();
     boolean makeVrReady = uriProjectLoader.shouldMakeVrReady();
-
-    uriProjectLoader = null;
-    activity.cancel();
+    boolean isDefaultBackup = uriProjectLoader.isDefaultBackup();
 
     unloadableFiles.add(projectFile.getName());
 
-    if (isBackup && isLoadingBackups) {
-      File mainProject = getMainProjectFile(projectFile);
+    // If this failed attempt was already a load of a backup, keep the existing value
+    // Otherwise, this is the main project, so set this to true
+    if (!isLoadingBackups) {
+      isMainProjectCorrupted = true;
+    }
 
-      LocalDateTime projectModifiedTime = FileUtilities.getModifiedDateTime(mainProject);
+    File mainProject = uriProjectLoader.getMainProjectFile();
+    LocalDateTime projectModifiedTime = FileUtilities.getModifiedDateTime(mainProject);
```

Comment:
Something changed the way this works.
The value of mainProject may be null, if the backup was not loaded, which causes getModifiedTime to throw and an uphelpful error dialog to show and then fall through to the main open window.
Previously there was no error and the user was given the chance to open earlier backups.
 
I think it was in the latest commit "Always show an asterisk in the title bar when a backup is open" 3d5c890e24c0bef2d05e4392ba7827bf5e869eb6